source/testing/index.md

```diff
@@ -1,79 +1,149 @@
-interacts with the system in the same way that it will within the context of the application,
+Rendering tests are especially useful for verifying the correct behavior of components and helpers.
+The component or helper interacts with the system in the same way that it will within the context of the application,
 including being rendered from a template and receiving Ember's lifecycle hooks.
 
-Examples of integration tests are:
+Examples of these kind of tests are:
 
 * An author's full name and date are properly displayed in a blog post.
 * A user is prevented from typing more than 50 characters into post's title field.
 * Submitting a post without a title displays a red validation state on the field and gives the user text indicating that the title is required.
 * The blog post list scrolls to position a new post at the top of the viewport.
 
+We can create an isolated rendering test with the `setupRenderingTest` helper.
+You can read more about it in the [Testing Components] or the [Testing Helpers] section.
+
+### Container Tests
+
+Container tests are useful if you are testing a small part of your app which doesn't require user interaction to be tested
+but which requires the application's container to be setup.
+If you would like to test a functionality regarding an Ember class instance,
+e.g. a controller, service or route, you can create this kind of test to verify its behavior.
+
+Some specific examples of these type of tests are:
+
+* A `fullName` attribute on a controller is computed by combining its `firstName` and `lastName` attributes.
+* The serializer properly converts the blog request payload into a blog post model object.
+* Blog dates are properly formatted through a `time` service.
+
+We can create these isolated container tests with the `setupTest` helper.
+You can read more about these type of tests in the [Testing Routes] and [Testing Controllers] section.
+
+### Simple Unit Tests
+
+If you would like to test an isolated, small-scoped functionality you can create a unit test.
+Unit tests are the go-to testing type, if you neither require user interactions nor an application container setup to create your test case.
+
+Some specific examples of these type of tests are:
+
+* The return value of a `getFullName` utility combines its `firstName` and `lastName` parameters correctly.
+* A computed property macro formats a price depending on its `currency` and `cents` dependent keys.
+* A helper computes the current date properly.
+
+We can create these type of isolated tests which neither require the application's container to be setup
+nor any user interaction with the `setupTest` helper as well.
+
 ### Testing Frameworks
 
 [QUnit] is the default testing framework for this guide, but others are supported through third-party addons.
 
+### Testing Blueprints
+
+By default whenever you are creating a new component, helper, service or any another module in your app,
```

Comment:
comma after service (just an Oxford comma - it's grammatically correct either way)